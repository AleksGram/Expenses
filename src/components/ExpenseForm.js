import React, { Component } from 'react';

export default class ExpenseForm extends Component {
    state = {
        description: '',
        note: '',
        amount: ''
    };

    onDescriptionChange = (e) => {
        this.setState({description: e.target.value});
    }

    onTextAreaChange = (e) => {
        this.setState({note: e.target.value});
    }
    
    onAmountChange = (e) => {
        const amount = e.target.value;

        if(amount.match(/^\d*(\.\d{0,2})?$/)) {
            this.setState({amount: amount})
        }
    }

    render () {
        return (
            <div>
                <form>
                    <input
                        type='text'
                        placeholder="Description"
                        autoFocus
                        value={this.state.description}
                        onChange={this.onDescriptionChange}
                    />
                    <input
                        type='text'
                        placeholder="Amount"
                        value={this.state.amount}
                        onChange={this.onAmountChange}
                    />
                    <textarea
                        placeholder="Add a note for your expense (optinal)"
                        value={this.state.note}
                        onChange={this.onTextAreaChange}
                    >
                    </textarea>
                    <button>Add Expense</button>
                </form>
            </div>
        )
    }
}

