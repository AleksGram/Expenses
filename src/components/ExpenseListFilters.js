import React, { Component } from 'react';
import { connect } from 'react-redux';
import { setTextFilter } from '../actions/filters';
import { sortByDate, sortByAmount, setStartDate, setEndDate } from '../actions/filters';
import { DateRangePicker } from 'react-dates';


export class ExpenseListFilters extends Component {
    constructor(props) {
        super(props);
        this.state = {
            calendarFocused: null
        }
    }
    onDatesChange = ({ startDate, endDate }) => {
        this.props.setStartDate(startDate);
        this.props.setEndDate(endDate);
    }

    onFocusChange = (calendarFocused) => {
        this.setState({ calendarFocused })
    }

    onTextChange = (e) => {
        this.props.setTextFilter(e.target.value)
    }

    onSortChange = (e) => {
        switch (e.target.value) {
            case 'amount':
                this.props.sortByAmount();
                break;
            case 'date':
                this.props.sortByDate();
                break;
        }
    }

    render() {
        return (
            <div className="content-container">
                <div className="input-group">
                    <div className="input-group__item">
                        <input
                            placeholder="Search expenses" 
                            className="text-input" 
                            type='text' 
                            value={this.props.filters.text} 
                            onChange={this.onTextChange} 
                        />
                    </div>
                    <div className="input-group__item">
                        <select
                            className="select"
                            value={this.props.filters.sortBy}
                            onChange={this.onSortChange}>
                            <option value="date">Date</option>
                            <option value="amount">Amount</option>
                        </select>
                    </div>
                    <div className="input-group__item">
                        <DateRangePicker
                            startDate={this.props.filters.startDate}
                            endDate={this.props.filters.endDate}
                            onDatesChange={this.onDatesChange}
                            focusedInput={this.state.calendarFocused}
                            onFocusChange={this.onFocusChange}
                            showClearDates={true}
                            numberOfMonths={1}
                            isOutsideRange={() => false}
                        />
                    </div>
                </div>
            </div>
        )
    }
}

const mapStateToProps = (state) => {
    return {
        filters: state.filters
    };
};

const mapDispatchToProps = (dispatch) => ({
    setStartDate: (startDate) => dispatch(setStartDate(startDate)),
    setEndDate: (endDate) => dispatch(setEndDate(endDate)),
    setTextFilter: (value) => dispatch(setTextFilter(value)),
    sortByAmount: () => dispatch(sortByAmount()),
    sortByDate: () => dispatch(sortByDate()),
})

export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters);