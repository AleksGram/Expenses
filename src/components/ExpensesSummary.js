import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import selectTotalExpenses from './../selectors/expenses-total';
import selectExpenses from './../selectors/expenses';
import numeral from 'numeral';



export const ExpensesSummary = ({ expenseCount, expenseSummary, expenses }) => {
    const description = expenseCount === 1 ? 'expense' : 'expenses';
    const formatTotalVal = numeral(expenseSummary / 100).format('$0,0.00');
    const difExpenses = expenses.length - expenseCount;

    const totalExpenses = (difExpenses > 0) ? `Expenses out of the filter: ${difExpenses}` : null;

    return (
        <div className="page-header">
            <div className="content-container">
                <h1 className="page-header__title">Viewing <span>{expenseCount}</span> {description} totaling <span>{formatTotalVal}</span> </h1>
                <h2 className="page-header__title">{totalExpenses}</h2>
                <div className="page-header__actions">
                    <Link className="button" to="/create" >Add Expense</Link>
                </div>
            </div>
        </div>
    )
}

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters)
    console.log(state.expenses);

    return {
        expenseCount: visibleExpenses.length,
        expenseSummary: selectTotalExpenses(visibleExpenses),
        expenses: state.expenses
    }
};

export default connect(mapStateToProps)(ExpensesSummary);


