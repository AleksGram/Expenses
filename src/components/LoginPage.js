import React, { Component } from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

export const LoginPage = ({ startLogin }) => {
    return (
        <div className="box-layout">
            <div className="box-layout__box">
                <h1 className="box-layout__title">Welcome to Expensify App</h1>
                <p>Controll your expensies easy</p>
                <button onClick={startLogin}>Login</button>
            </div>
        </div>
    )
}

const mapDispatchToProps = (dispatch) => ({
    startLogin: () => dispatch(startLogin())
})

export default connect(null, mapDispatchToProps)(LoginPage);