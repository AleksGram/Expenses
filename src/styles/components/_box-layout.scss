.box-layout {
    background: url("/images/money.jpg")  ;
    height: 100vh;
    width: 100vw;
    -webkit-background-size: cover;
    -moz-background-size: cover;
    -o-background-size: cover;
    background-size: cover;
    position: fixed;
    top: 0;
    left: 0;
    display: flex;
    justify-content: center;
    align-items: center;
}

.box-layout__box {
    background: fade-out(#00a699, .15);
    border-radius: 3px;
    text-align: center;
    width: 25rem;
    padding: $l-size $m-size;
}

.box-layout__title {
    margin: 0 0 $m-size 0;
    line-height: 1;
}