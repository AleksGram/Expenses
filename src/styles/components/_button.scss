.button {
    background: $b-green;
    border: none;
    font-weight: 3000;
    line-height: 1;
    display: inline-block;
    padding: $s-size;
    font-size: $font-size-large;
    text-decoration: none;
}

.button--link {
    background: transparent;
    color: $l-green;
}