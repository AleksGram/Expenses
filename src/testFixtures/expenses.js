import moment from 'moment';

export default [{
    id: '1',
    description: 'Car',
    note: 'new',
    amount: 5000,
    createdAt: 0
},
{
    id: '2',
    description: 'Wash',
    note: '',
    amount: 50,
    createdAt: moment(0).subtract(4, 'days').valueOf()
},
{
    id: '3',
    description: 'Gym',
    note: 'membership',
    amount: 100,
    createdAt: moment(0).add(4, 'days').valueOf()
}]
