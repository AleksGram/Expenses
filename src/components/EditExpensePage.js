import React, { Component } from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { editExpense } from '../actions/expenses';
import { removeExpense } from '../actions/expenses';


export class EditExpensePage extends Component {
  constructor (props) {
    super(props)
    this.onSubmit = this.onSubmit.bind(this);
    this.onClick = this.onClick.bind(this)

  }
  onSubmit = (expense) => {
    this.props.editExpense(this.props.expense.id, expense)
    this.props.history.push('/');
  }
  onClick = () => {
    this.props.removeExpense({id: this.props.expense.id})
    this.props.history.push('/')

  }

  render () {
    return (
      <div>
      <ExpenseForm
        expense = {this.props.expense}
        onSubmit={this.onSubmit}
      />
      <button 
      onClick={this.onClick}
      >Remove</button>  
    </div>
    )
  }
}

// const EditExpensePage = (props) => {
//   console.log(props);
//   return (
//     <div>
//       <ExpenseForm
//         expense = {props.expense}
//         onSubmit={(expense) => {
//           props.dispatch(editExpense(props.expense.id, expense))
//           props.history.push('/');
//         }}
//       />
//       <button 
//       onClick={
//         () => {
//           props.dispatch(removeExpense({id: props.expense.id}))
//           props.history.push('/')
//         }
//       }
//       >Remove</button>  
//     </div>
//   );
// };



const mapStateToPros = (state, props) => {
  return {
    expense: state.expenses.find((expense) => expense.id === props.match.params.id)
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    editExpense: (expense) => dispatch(editExpense(this.expense.id, expense)),
    removeExpense: () => dispatch(removeExpense({id: this.props.expense.id}))
  }
}

export default connect(mapStateToPros, mapDispatchToProps)(EditExpensePage);
