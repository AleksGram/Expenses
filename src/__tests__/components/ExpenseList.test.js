import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseList } from '../../components/ExpenseList';
import expenses from '../../testFixtures/expenses';


test('renderExpenseList', () => {
    const wrapper = shallow(<ExpenseList expenses={expenses}/>);

    expect(wrapper).toMatchSnapshot();
})

test('renderExpenseList_noExpenses', () => {
    const wrapper = shallow(<ExpenseList expenses={[]}/>);

    expect(wrapper).toMatchSnapshot();
})