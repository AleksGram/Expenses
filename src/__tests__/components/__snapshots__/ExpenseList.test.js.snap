// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderExpenseList 1`] = `
<div>
  <h1>
    Expense List
  </h1>
  <ExpenseListItem
    amount={5000}
    createdAt={0}
    description="Car"
    id="1"
    key="1"
    note="new"
  />
  <ExpenseListItem
    amount={50}
    createdAt={-345600000}
    description="Wash"
    id="2"
    key="2"
    note=""
  />
  <ExpenseListItem
    amount={100}
    createdAt={345600000}
    description="Gym"
    id="3"
    key="3"
    note="membership"
  />
</div>
`;

exports[`renderExpenseList_noExpenses 1`] = `
<div>
  <h1>
    Expense List
  </h1>
  <p>
    No expenses
  </p>
</div>
`;
